### DISCUSSION TITLE
"task queue is full" ERROR

### DISCUSSION THREAD

**Discussion by hk8805** [milvus-io/milvus#24298] [Q&A and General discussion]
when the request for Milvus-cluster service double, I get a error "task queue is full" .
is there a config to set the queue limits?

**Comment by yhmo** [comment]
The task queue of proxy is defined in milvus.yaml:
```
proxy:
  maxTaskNum: 1024 # max task number of proxy task queue
```

Task queue limit of query node and index node is hard-coded in source code with value 1024.

When this error occurs, that means the system cannot afford too many tasks with current hardware resource. You might need to increase the cpu cores.

**Comment by xiaofan-luan** [comment]
usually task queue full means milvus can not handle concurrent requests.
You may want to add querynode resources?
